app/src/main/cpp/skyline/common.h

```diff
@@ -338,6 +339,18 @@ namespace skyline {
         constexpr std::size_t Hash(std::string_view view) {
             return frz::elsa<frz::string>{}(frz::string(view.data(), view.size()), 0);
         }
+
+        /**
+         * @brief Fills an array with random data from a Mersenne Twister pseudo-random generator
+         * @note Currently the generator is not seeded so produced values will be identical between two runs
+         */
+        template<typename T, typename S = u64> requires (std::is_integral_v<T> && std::is_integral_v<S>)
```

Comment:
Usage of `T` here is going to be problematic since you'll have multiple `std::mt19937_64` generators based on the type. You'd want to move the `generator` outside of the function (in a `util::detail` namespace) so that you only have one generator per-thread that multiple `std::uniform_int_distribution`s can utilize.